backend/src-common/src/main/java/com/siemens/sw360/datahandler/db/ComponentDatabaseHandler.java

```diff
@@ -646,10 +646,12 @@ public ReleaseClearingStateSummary getReleaseClearingStateSummary(Set<String> id
         return out;
     }
 
-    private Optional<ReleaseLink> createReleaseLink(Set<String> visitedIds, Map.Entry<String, ?> entry, String id, Release release, Map<String, Release> releaseMap, String parentId) {
-        if (visitedIds.add(id)) {
+    private Optional<ReleaseLink> createReleaseLink(Stack<String> visitedIds, Map.Entry<String, ?> entry, String id, Release release, Map<String, Release> releaseMap, String parentId) {
+        ReleaseLink releaseLink = null;
```

Comment:
Prompted by your comment, I reviewed argument lists in all four related methods and cleaned up/reorganized them a little. See
[60f788e](https://github.com/sw360/sw360portal/pull/264/commits/60f788efda19d5bbb3c8456e901e91193a7713ee)